pkg/fixtures/fixtures.go

```diff
@@ -118,6 +120,21 @@ func (fxt *Fixture) Execute() error {
 	return nil
 }
 
+func errWasExpected(err error, expectedErrorType string) bool {
+	if rerr, ok := err.(requests.RequestError); ok {
+		if e, ok := rerr.Body["error"].(map[string]interface{}); ok {
+			if t, ok := e["type"]; ok && t == expectedErrorType {
```

Comment:
I wonder if there is a way to simplying this instead of having to next 3 levels of ifs just to deconstruct our error. This diff introduces `RequestError` for its purpose so we can probably make it in a way to makes it easier.

Reply:
so I got a little excited to try and figure out how to do that in Go, here's an idea of how to clean that up a bit.

1. We create structs to represent the expected _error_ response from the API.
```
// this could also be RequestBody and have fields for both error and success responses but not sure this is needed
type RequestErrorBody struct {
    Error RequestError `json:"error"`
}

type RequestError struct {
    Message string `json:"message"`
    Code string `json:"code"`
}
```

2. Now we can Marshal `body` into a `RequestErrorBody` and have all field natively accessible
```
var reqErrorBody RequestErrorBody
json.Unmarshal(body, &reqErrorBody)
```
now we can check if reqErrorBody has the correct code
```
if err != nil && !err.Error.Code != data.ExpectedErrorType {
  return err
}
```

Reply:
Thanks for demoing this to us, I'll incorporate it into my next change!